internal/scylla/client.go

```diff
@@ -136,11 +136,65 @@ func (c *Client) doHttpRequest(req *http.Request) (resp *http.Response, temporar
 		return
 	}
 
-	temporaryErr = resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusGatewayTimeout || resp.StatusCode == http.StatusTooManyRequests
+	body, err := getBody(resp)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to get response body: %w", err)
+	}
+
+	body2, err := getBody(resp)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to get response body: %w", err)
+	}
+
+	code := getErrorCode(body)
+
+	tflog.Trace(req.Context(), "getErrorCode: "+code, map[string]interface{}{
+		"code":   resp.StatusCode,
+		"status": resp.Status,
+		"body1":  string(body),
+		"body2":  string(body2),
+	})
+
+	if code == "000001" {
+		return resp, true, fmt.Errorf("too many request response")
+	}
+	temporaryErr = temporaryErr || resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusGatewayTimeout || resp.StatusCode == http.StatusTooManyRequests
 	return
 }
 
+func getBody(resp *http.Response) ([]byte, error) {
+	if resp.Body == nil || resp.Body == http.NoBody {
+		return nil, nil
+	}
+	defer resp.Body.Close()
+	var buf bytes.Buffer
+	if _, err := buf.ReadFrom(resp.Body); err != nil {
+		return nil, err
+	}
+	resp.Body = io.NopCloser(bytes.NewReader(buf.Bytes()))
+	return buf.Bytes(), nil
+}
+
+func getErrorCode(body []byte) string {
+	if len(body) == 0 {
+		return ""
+	}
+
+	var data = struct {
+		Error string `json:"error"`
+	}{}
+
+	if err := json.Unmarshal(body, &data); err != nil {
+		// Not every response carry json, so we return empty string here.
+		return ""
```

Comment:
1. returning empty string is wrong, it would be way more useful to return string(body) to display raw payload as error (in case of various "500 Internal Error" text/plain payloads)
2. Please return or at very very minimum log the error, so it is obvious something went wrong, that we received wrongly encoded response and so on.

Reply:
Don't see any point in doing so, but I have added it